src/nle.c

```diff
@@ -26,8 +26,68 @@
 
 extern int unixmain(int, char **);
 
+short
+vt_font_attr_convert(TMTCHAR *c)
+{
+    /* short = 2x 1bit pad, 6x 1bit for each bool, 2x 4bit for each enum (1-9)
+     */
+    short font = 0;
+    font |= (c->a.bold << 13);
+    font |= (c->a.dim << 12);
+    font |= (c->a.underline << 11);
+    font |= (c->a.blink << 10);
+    font |= (c->a.reverse << 9);
+    font |= (c->a.invisible << 8);
+    font |= ((c->a.fg & 0xF) << 4);
+    font |= ((c->a.bg & 0XF) << 0);
+    return font;
+}
+
+void
+nle_vt_callback(tmt_msg_t m, TMT *vt, const void *a, void *p)
+{
+    const TMTSCREEN *s = tmt_screen(vt);
+
+    nle_ctx_t *nle = (nle_ctx_t *) p;
+    if (!nle || !nle->observation) {
+        return;
+    }
+
+    switch (m) {
+    case TMT_MSG_BELL:
+        break;
+
+    case TMT_MSG_UPDATE:
+        for (size_t r = 0; r < s->nline; r++) {
+            if (s->lines[r]->dirty) {
+                for (size_t c = 0; c < s->ncol; c++) {
+                    size_t offset = (r * NLE_TERM_CO) + c;
+                    TMTCHAR *tmt_c = &(s->lines[r]->chars[c]);
+
+                    if (nle->observation->terminal_chars) {
+                        nle->observation->terminal_chars[offset] = tmt_c->c;
+                    }
+
+                    if (nle->observation->terminal_fonts) {
+                        nle->observation->terminal_fonts[offset] =
+                            vt_font_attr_convert(tmt_c);
+                    }
+                }
+            }
+        }
+        tmt_clean(vt);
+        break;
+
+    case TMT_MSG_ANSWER:
+        break;
+
+    case TMT_MSG_MOVED:
```

Comment:
This reminds me: Cursor position is something we'll likely use in the ttyrecs. It might be a good idea to include tty cursor position in the observations here, too. Either as 2-dim short `tty_cursor` array, or perhaps we create a new "other interesting stuff" observation -- something that the kitchen-sink `internal` does right now, which now also contains `uhunger` etc. We could refactor that to be a bit less random?

Reply:
I agree, adding the cursor position should be quite useful.

Reply:
added a cursor observation and update it on the TMT_MSG_MOVED case, which corresponds to cursor move. I haven't included a toggle display bool though, as per heiner's comment.